pre_processing/read_seviri.F90

```diff
@@ -435,7 +443,7 @@ subroutine read_seviri_l1_5_nat_or_hrit(l1_5_file, imager_geolocation, &
    column0 = startx - 1
    column1 = startx - 1 + nx - 1
 
-   if (.not. hrit_proc .or. (nx .eq. 3712 .and. ny .eq. 3712)) then
+   if (.not. hrit_proc .and. (nx .eq. 3712 .and. ny .eq. 3712)) then
```

Comment:
Why has this been changed?

Reply:
The original code required all files that did not contain "H-000-MSG" to be processed using the memory efficient form. That is incompatible with the angle subsetting. By switching to `.and.` we only use the memory efficient mode for full-disc reads that don't contain that string.

Reply:
I'll look into this more, I don't think it's doing what you expect it to. Or maybe it's not doing what I expect it to!

Reply:
This still is problematic, as it won't work in memory efficient mode for RSS images that have 1392 rows for VIS/IR channels, nor will it work for super RSS data that has differing row and columns counts.
Can this be refactored to take the above into account? We know the rows + columns requested by the user and we also (theoretically, even though it's broken just now) know the actual rows + columns in the data, so it should be a case of comparing the two.

Reply:
(a) Please email me a path to one of these RSS images. They clearly matter to you and so should be part of my regression tests.
(b) Could you explain what HRIT means?
(c) My issue with the line is that if HRIT is false, which it is for the file I use in my regression test, it uses the SEVIRI reader exclusively. We believe that that is presenting an angle the wrong way around and so cannot be subset correctly.
(d) The refactor would be a clause looking for subsetting. I'd prefer the SEVIRI reader be corrected, but I can't do that.